File: src/main/java/com/sara/student/Student.java
package com.sara.student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest() {
//        int max = ;
//        if (english > math) {
//            max = english;
//        } else {
//            max = math;
//        }
        return (english > math) ? english : math;
    }

    public void print() {
        System.out.print(name + "\t" + english + "\t" + math +
                "\t" + getAverage());
        if (getAverage() >= 60) {
            System.out.print("\tPASS");
        } else {
            System.out.print("\tFAILED");
        }

        int average = getAverage();
        char grading = 'F';
        switch (average/10) {
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'F';
        }
        System.out.println("\t" + grading);
    }

    public int getAverage() {
        return (english+math)/2;
    }
}

File: src/main/kotlin/com/sara/student/Student.kt
package com.kotlin

import java.util.*

fun main(args: Array<String>) {
//    userInput()
    val stu = Student("Sara", 80, 90)
    stu.print()
    val test = 123
    println("Test is $test")
    println("High score: ${stu.highest()}")
}

class Student(var name: String?, var english: Int, var math: Int){
    fun print(){
        println(name + "\t" + english + "\t" + math +
                "\t" + getAverage() + "\t" +
                if (getAverage() >= 60) "PASS" else "FAILED"
        )
    }

    fun getAverage() : Int {
        return (english+math)/2
    }

    fun highest() : Int {
        var max = if (english > math) {
            println("english")
            english
        } else {
            println("math")
            math
        }
//        if (english > math) {
//            max = english
//        } else {
//            max = math
//        }
        return max
    }

    fun nameCheck(){
        println(name?.length)
    }
}

private fun userInput() {
    val scanner = Scanner(System.`in`)
//    print("Please enter student's name: ")
//    var name = scanner.next()
    var name = null
    print("Please enter student's english: ")
    var english = scanner.nextInt()
    print("Please enter student's math: ")
    var math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
    stu.nameCheck()
}
